### ISSUE TITLE
[Bug]: Notifications Trigger Too Early Despite Retry Configuration

### ISSUE THREAD

**Issue by Osvaldo-Net** [operacle/checkcle#180]
### Environment

 Ubuntu 24.04.3 LTS

### Bug Description

I have configured the system to mark services as inactive after 3 retry attempts. However, in the application, they do not appear as down, even though notifications are still sent after just one failed ping, which is unnecessarily demanding for the system.

### Steps to Reproduce

To reproduce the issue, simply create a monitor, configure the alert, and set the retry attempts to one, two, or three the notification will still be sent on the first ping failure.

### Expected Behavior

It would be ideal if the notification were only sent once the host is actually marked as inactive in the tool, i.e., after 3 failed attempts, to prevent false positives.

### Screenshots or Videos

<img width="802" height="221" alt="Image" src="https://github.com/user-attachments/assets/d43c8a01-7568-4e13-a4a8-4c92b42d6aa5" />

### Logs

```shell

```

### Checklist

- [x] I have searched existing issues to avoid creating duplicates.
- [x] I have provided sufficient information to reproduce the issue.

**Comment by tolaleng** [comment]
Hello @Osvaldo-Net 

Appreciate your feedback! I’ll be updating this feature to improve the experience.

**Comment by Osvaldo-Net** [comment]
I might have overlooked something, but this behavior seems related to this feature:
[Feature]: Add configurable notification delay for downtime alerts [#176]
